contrib/zenazn/goji/web/example_test.go

```diff
@@ -0,0 +1,27 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package web_test
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/zenazn/goji"
+	"github.com/zenazn/goji/web"
+	webtrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/zenazn/goji/web"
+)
+
+func ExampleMiddleware() {
+	// Using the Router middleware lets the tracer determine routes for
+	// use in a trace's resource name ("GET /user/:id")
+	// Otherwise the resource is only the method ("GET", "POST", etc.)
+	goji.Use(goji.DefaultMux.Router)
```

Comment:
All of the examples goji gives for running a service use the `goji.DefaultMux` router. Unfortunately I haven't found a way around it.

This step is not necessary but adds extra information (route names) if it is done. As far as I can tell, the router you need to use is the one that belongs to whatever `Mux` you are using. (https://godoc.org/github.com/zenazn/goji/web#Mux.Router). In this case, that's DefaultMux. You're right it's exported and can be set, but it's a struct, not an interface type, so it's somewhat safe.

You can see in the docs that there are other implications of using the `Router` middleware. Maybe we want to just skip mentioning it. I added it here because it adds useful info to the traces, but I don't want to make examples confusing.

Reply:
> All of the examples goji gives for running a service use the goji.DefaultMux router. 

Sure. Let's keep it. I was just making sure :)